jest.config.js

```diff
@@ -1,13 +1,9 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
 /* eslint-disable import/no-commonjs */
+const defaultConfig = require('./jest.base.config')
+
 module.exports = {
-  moduleNameMapper: {
-    '^dnd-core$': 'dnd-core/dist/cjs',
-    '^react-dnd$': 'react-dnd/dist/cjs',
-    '^react-dnd-html5-backend$': 'react-dnd-html5-backend/dist/cjs',
-    '^react-dnd-touch-backend$': 'react-dnd-touch-backend/dist/cjs',
-    '^react-dnd-test-backend$': 'react-dnd-test-backend/dist/cjs',
-    '^react-dnd-test-utils$': 'react-dnd-test-utils/dist/cjs',
-    '\\.(css|less|eot|svg|ttf|woff|woff2)$': '<rootDir>/jest.styleMock.js',
-  },
-  transformIgnorePatterns: ['/node_modules/(?!(react-syntax-highlighter)/)'],
+  ...defaultConfig,
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
+  testEnvironment: 'jsdom-sixteen',
```

Comment:
Why do we need this (over the default jsdom environment that jest provides)?

Reply:
Otherwise I get the following error:

```
Error: Uncaught [TypeError: window.getSelection is not a function]
          at reportException (/home/kulla/workspace/serlo/edtr-io/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:62:24)                                                         
          at innerInvokeEventListeners (/home/kulla/workspace/serlo/edtr-io/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:332:9)                                                     
          at invokeEventListeners (/home/kulla/workspace/serlo/edtr-io/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:267:3)                                                          
          at HTMLUnknownElementImpl._dispatch (/home/kulla/workspace/serlo/edtr-io/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:214:9)
...
```
Here I used the solution https://stackoverflow.com/questions/43677034/enzyme-jest-window-getselection-does-not-work/63064633#63064633

However jest seems to use JSDOM 16 since version 26 which we already have. I have no clue why it does not work already...

Reply:
@inyono For now I give up on this issue. Even though we have stated `jest: "^26.0.0"` in our package.json the v25 seems to be around anyhow:

```
$ ▶ yarn why jest-environment-jsdom                                                                               
yarn why v1.22.4
[1/4] Why do we have the module "jest-environment-jsdom"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "jest-environment-jsdom@25.5.0"
info Reasons this module exists
   - "_project_#jest-config" depends on it
   - Hoisted from "_project_#jest-config#jest-environment-jsdom"
info Disk size without dependencies: "436KB"
info Disk size with unique dependencies: "4.99MB"
info Disk size with transitive dependencies: "28.72MB"
info Number of shared dependencies: 117
=> Found "jest#jest-environment-jsdom@26.5.2"
info Reasons this module exists
   - "_project_#jest#@jest#core#jest-config" depends on it
   - Hoisted from "_project_#jest#@jest#core#jest-config#jest-environment-jsdom"
...
```

I guess we need to fix this together...

Reply:
Seems like the issue is `tsdx` which has `jest@25` as a dependency. Since I failed to get `yarn` or `jest` to use the newer version, we can just ignore the issue.